### PR TITLE
Implement etcd based minion data cache

### PR DESCRIPTION
**Pull request by FedericoCeratto**
### What does this PR do?
Implement a minion data cache plugin for etcd.

### Tests written?
No. Tested manually with a local DB instance. I'm open to suggestions to create integration tests.

### CODE REVIEW

File: salt/cache/etcd_cache.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+            'cert': __opts__.get('etcd.cert', None),
+            'ca_cert': __opts__.get('etcd.ca_cert', None),
+    }
+
+    global client, path_prefix
+    path_prefix = __opts__.get('etcd.path_prefix', _DEFAULT_PATH_PREFIX)
+    if path_prefix != "":
+        path_prefix = '/{0}'.format(path_prefix.strip('/'))
+    log.info("etcd: Setting up client with params: %r", etcd_kwargs)
+    client = etcd.Client(**etcd_kwargs)
```
Comment (cachedout) [inline salt/cache/etcd_cache.py]: Same issue here as in the other PR. We should not be doing this sort of initialization in `__virtual__` IMHO.

**Review by cachedout — CHANGES_REQUESTED** [review]

File: salt/cache/etcd_cache.py (hunk 1 above)
Comment (FedericoCeratto) [inline salt/cache/etcd_cache.py]: Updated.

**Comment by cachedout** [comment]
Go Go Jenkins!

File: salt/cache/etcd_cache.py (hunk 1 above)
Comment (FedericoCeratto) [inline salt/cache/etcd_cache.py]: Done

Comment (FedericoCeratto) [inline salt/cache/etcd_cache.py]: Done

**Review by cachedout — APPROVED** [review]

**Pull request merged**